### PR TITLE
parse and validate some input parameters as filesystem-paths

### LINKED ISSUE #648: Using --input to add resources on Windows causes yaml error
**What happened**:

When adding a Helm Chart resource to component archive using --input arguments, a error occurs:
```
processing resource (by options)...
Error: resource (by options): yaml: found unknown escape character
```

**What you expected to happen**:

The resource should be added to the component archive without errors.

**How to reproduce it (as minimally and precisely as possible)**:

Assuming you are in a directory: `E:\t\bugrepo`

```PowerShell
# Create component archive
ocm create ca --file ca --scheme ocm.software/v3alpha1 --provider test.com test.com/test 1.0.0

# Pull Helm chart (normally this will be a Helm chart that is build in a pipeline
helm pull oci://registry-1.docker.io/bitnamicharts/postgresql --version 14.0.5

# Add resource (notice the full path here, because that is causing the issue)
ocm add resources --file ca --name postgresql --type helmChart --inputType helm --inputPath E:\t\bugrepo\postgresql-14.0.5.tgz --inputVersion 14.0.5
```

**Anything else we need to know**:
The problem is the full path for the --inputPath argument. What happens:
* ocm creates a json representation of the input variables
* ocm runs the templater over this json, by default this is the subst templater, this templater is also used when specifying none)
* the subst templater removes the backslash escaping from the json causing the error
* the templater is applied at [line 57: https://github.com/open-component-model/ocm/blob/main/cmds/ocm/commands/ocmcmds/common/addhdlrs/utils.go](https://github.com/open-component-model/ocm/blob/main/cmds/ocm/commands/ocmcmds/common/addhdlrs/utils.go)

You can workaround this error if you use double backslashes: ` --inputPath E:\\t\\bugrepo\\postgresql-14.0.5.tgz`. This will lead to another error which I will describe in another issue.

**Environment**:

OS: Windows 11 Pro 23H2
ocm: v0.6.0

### PR DESCRIPTION
**Pull request by hilmarf**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Fixes #648

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,10 +37,10 @@ require (
 	github.com/imdario/mergo v0.3.16
 	github.com/klauspost/compress v1.17.2
 	github.com/klauspost/pgzip v1.2.6
-	github.com/mandelsoft/filepath v0.0.0-20230412200429-36b1eb66bd27
+	github.com/mandelsoft/filepath v0.0.0-20240223090642-3e2777258aa3
 	github.com/mandelsoft/logging v0.0.0-20230905123808-7042ee3aae45
 	github.com/mandelsoft/spiff v1.7.0-beta-5
-	github.com/mandelsoft/vfs v0.4.1
+	github.com/mandelsoft/vfs v0.4.3-0.20240223095430-def3828dd181
```
Comment (hilmarf) [inline go.mod:43]: @mandelsoft are you going to spend a new release for this?

File: go.mod [hunk 2]
```diff
@@ -37,10 +37,10 @@ require (
 	github.com/imdario/mergo v0.3.16
 	github.com/klauspost/compress v1.17.2
 	github.com/klauspost/pgzip v1.2.6
-	github.com/mandelsoft/filepath v0.0.0-20230412200429-36b1eb66bd27
+	github.com/mandelsoft/filepath v0.0.0-20240223090642-3e2777258aa3
```
Comment (hilmarf) [inline go.mod:40]: @mandelsoft are you going to spend a new release for this?

Comment (mandelsoft) [reply go.mod:40]: there will be a release for vfs

File: pkg/cobrautils/flag/path.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// Path defines a filepath flag with specified name, default value, and usage string.
+// The return value is the address of a string variable that stores the value of the flag.
+func Path(f *pflag.FlagSet, name string, value string, usage string) *string {
+	p := new(string)
+	PathVarP(f, p, name, "", value, usage)
+	return p
+}
+
+// PathP is like Path, but accepts a shorthand letter that can be used after a single dash.
+func PathP(f *pflag.FlagSet, name, shorthand string, value string, usage string) *string {
```
Comment (mandelsoft) [inline pkg/cobrautils/flag/path.go:54]: could you please  also add the VarPF variant

File: pkg/cobrautils/flag/path_array.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+// PathArray defines a filepath flag with specified name, default value, and usage string.
+// The return value is the address of a []string variable that stores the value of the flag.
+func PathArray(f *pflag.FlagSet, name string, value []string, usage string) *[]string {
+	p := []string{}
+	PathArrayVarP(f, &p, name, "", value, usage)
+	return &p
+}
+
+// PathArrayP is like PathArray, but accepts a shorthand letter that can be used after a single dash.
+func PathArrayP(f *pflag.FlagSet, name, shorthand string, value []string, usage string) *[]string {
```
Comment (mandelsoft) [inline pkg/cobrautils/flag/path_array.go:104]: and the same here

File: pkg/cobrautils/flag/replace.go [hunk 5]
```diff
@@ -31,3 +27,15 @@ func IntVarPF(f *pflag.FlagSet, p *int, name, shorthand string, value int, usage
 	f.IntVarP(p, name, shorthand, value, usage)
 	return f.Lookup(name)
 }
+
+// PathVarPF is like PathVarP, but returns the created flag.
+func PathVarPF(f *pflag.FlagSet, p *string, name, shorthand string, value string, usage string) *pflag.Flag {
```
Comment (mandelsoft) [inline pkg/cobrautils/flag/replace.go:32]: this location here is for additional flavors for the standard pflag option types.
For own ones, these flavors should directly be places in the appropriate source file.

**Review by mandelsoft — CHANGES_REQUESTED** [review]

**Pull request merged**